polar-core/src/resource_block.rs

```diff
@@ -298,18 +300,43 @@ impl ResourceBlocks {
         resource: Term,
         declarations: Declarations,
         shorthand_rules: Vec<ShorthandRule>,
-    ) {
-        self.declarations.insert(resource.clone(), declarations);
-        self.shorthand_rules
-            .insert(resource.clone(), shorthand_rules);
+    ) -> PolarResult<()> {
+        // Check for duplicate existing shorthand rules and return a
+        // ValidationError if we discover one
```

Comment:
Talked offline — we're gonna punt on the duplication check for this PR since it'd be best to implement it as a policy-wide check (i.e., not specific to the resource blocks feature).